cli/src/commands/extensions/api.rs

```diff
@@ -381,69 +443,125 @@ async fn parse_lockfile(
 #[op]
 #[cfg(unix)]
 fn run_sandboxed(process: Process) -> Result<ProcessOutput> {
-    // Setup process to be run.
-    let mut command = Command::new(&process.cmd);
-    command.args(&process.args);
-    command.stdin(process.stdin);
-    command.stdout(process.stdout);
-    command.stderr(process.stderr);
-
-    // Apply sandbox to subprocess after fork.
-    unsafe {
-        command.pre_exec(move || {
-            fn into_ioerr<E: Into<Box<dyn std::error::Error + Send + Sync>>>(err: E) -> io::Error {
-                io::Error::new(io::ErrorKind::Other, err)
-            }
-
-            let home_dir = dirs::home_dir().map_err(into_ioerr)?;
-
-            let mut birdcage = if process.exceptions.strict {
-                Birdcage::new().map_err(into_ioerr)?
+    let mut stdout_fds: ProcessStdioFds = process.stdout.try_into()?;
+    let mut stderr_fds: ProcessStdioFds = process.stderr.try_into()?;
+
+    match unsafe { libc::fork() } {
+        -1 => Err(io::Error::last_os_error().into()),
+        // Handle child process.
+        0 => {
+            // Connect STDOUT/STDERR with parent if necessary.
+            stdout_fds.replace_fd(libc::STDOUT_FILENO)?;
+            stderr_fds.replace_fd(libc::STDERR_FILENO)?;
+
+            // Apply sandboxing rules.
+            lock_process(process.exceptions)?;
+
+            // Setup process to be run.
+            let mut command = Command::new(&process.cmd);
+            command.args(&process.args);
+            command.stdin(process.stdin);
+            command.stdout(Stdio::inherit());
+            command.stderr(Stdio::inherit());
+
+            // Wait for process to complete.
+            let status = command.status()?;
```

Comment:
It should be fine considering that process is what everyone is waiting on, right? Are you concerned about the child not being able to write to stdout because the pipe is still open?